frontera/contrib/backends/sqlalchemy/components.py

```diff
@@ -123,18 +123,18 @@ def frontier_stop(self):
 
     @retry_and_rollback
     def fetch(self, fingerprints):
-        to_fetch = [f for f in fingerprints if f not in self._cache]
+        to_fetch = [to_native_str(f) for f in fingerprints if f not in self._cache]
         self.logger.debug("cache size %s", len(self._cache))
         self.logger.debug("to fetch %d from %d", len(to_fetch), len(fingerprints))
 
         for chunk in chunks(to_fetch, 128):
             for state in self.session.query(self.model).filter(self.model.fingerprint.in_(chunk)):
-                self._cache[str(state.fingerprint)] = state.state
+                self._cache[to_bytes(state.fingerprint)] = state.state
```

Comment:
And here you store bytes to `self._cache`.

please also check if it fits the code in parent class https://github.com/scrapinghub/frontera/blob/master/frontera/contrib/backends/memory/__init__.py#L122
I see there unicode requested from meta for PY3.